elasticdl/python/collective_ops/communicator.py

```diff
@@ -57,15 +63,28 @@ def allreduce(self, data, op="MEAN"):
             )
             return CollectiveCommunicatorStatus.FAILED, data
         if self._ftlib is not None:
-            res = self._ftlib.wait_gradients_ready(data)
-            if res == FTAllReduceStatus.SUCCESS:
+            res = self._ftlib.wait_gradients_ready(params=data)
+            if (
+                res == CommLibStatus.SUCCESS
+                or res == FTAllReduceStatus.NO_NEED
+            ):
                 return CollectiveCommunicatorStatus.SUCCEEDED, data
             else:
                 return CollectiveCommunicatorStatus.FAILED, data
         else:
             logger.warning(_FTLIB_UNINSTALLED_DEFAULT_STATUS_MESSAGE)
             return CollectiveCommunicatorStatus.SUCCEEDED, data
 
+    def tf_broadcast(self, params, src_rank):
```

Comment:
Instead of exposing a new function, we can check the data type and then merge the logic into `broadcast()`?